homeassistant/components/velux/config_flow.py

```diff
@@ -43,13 +54,59 @@ async def async_step_user(
                 errors["base"] = "unknown"
             else:
                 return self.async_create_entry(
-                    title=user_input[CONF_HOST],
-                    data=user_input,
+                    title=self.context.get("unique_id") or user_input[CONF_HOST],
+                    data={**self.discovery_data, **user_input},
                 )
 
-        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        data_schema = self.discovery_schema or self.add_suggested_values_to_schema(
+            DATA_SCHEMA, user_input
+        )
         return self.async_show_form(
             step_id="user",
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle discovery by DHCP."""
+        return await self._async_handle_discovery(
+            {
+                CONF_HOST: discovery_info.ip,
+                CONF_MAC: format_mac(discovery_info.macaddress),
+                CONF_NAME: discovery_info.hostname.upper().replace("LAN_", ""),
+            }
+        )
+
+    async def _async_handle_discovery(
+        self, discovery_info: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a discovered Velux device."""
+        # The hostname ends with the last 4 digits of the device MAC address.
+        self.discovery_data[CONF_HOST] = discovery_info[CONF_HOST]
+        if discovery_info[CONF_MAC]:
+            self.discovery_data[CONF_MAC] = discovery_info[CONF_MAC]
+        self.discovery_data[CONF_NAME] = discovery_info[CONF_NAME]
+
+        await self.async_set_unique_id(discovery_info[CONF_NAME])
+        self._abort_if_unique_id_configured(updates=self.discovery_data)
+
+        # Abort if config_entry already exists without unigue_id configured.
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if (
+                entry.data[CONF_HOST] == discovery_info[CONF_HOST]
+                and entry.unique_id is None
+                and entry.state == ConfigEntryState.LOADED
+            ):
+                self.hass.config_entries.async_update_entry(
+                    entry=entry,
+                    unique_id=discovery_info[CONF_NAME],
+                    data={**entry.data, **self.discovery_data},
+                )
+                return self.async_abort(reason="already_configured")
+
+        self.discovery_schema = self.add_suggested_values_to_schema(
+            DATA_SCHEMA, discovery_info
+        )
+        return await self.async_step_user()
```

Comment:
I think it would make sense for the discovery flow to not even show the option to change the host. We already know that its valid, otherwise it would not advertise from it.

Reply:
I changed the discovery_schema and included a SelectSelector with custom_value=False. That way the user cannot change the host value and I do not need to create a new step for a different Form. 
Moreover with the SelectOptionDict I provide a label which in the UI shows a string of "Hostname (IP Address)".  The Hostname is printed on the label of the KLF200, so the user do not need to know the IP, just in case he has more than one device.